src/sentry/api/serializers/models/team.py

```diff
@@ -88,6 +98,24 @@ def get_access_requests(item_list: Sequence[Team], user: User) -> AbstractSet[Te
     return frozenset()
 
 
+class _TeamSerializerResponseOptional(TypedDict, total=False):
+    externalTeams: list[ExternalActorResponse]
+    organization: OrganizationSerializerResponse
+    projects: ProjectSerializerResponse
```

Comment:
What happens to these references during runtime?

Reply:
per [PEP-563](https://www.python.org/dev/peps/pep-0563), when using `from __future__ import annotations`, type annotations are no longer evaluated at runtime, so they are simply strings.

This does cause some issues for our API docs generating, as it does happen at runtime instead of statically, so `get_type_hint` will fail to resolve the string. https://github.com/getsentry/sentry/pull/31958 attempts to address this.